src/dbg/debugger.cpp

```diff
@@ -581,7 +583,42 @@ static void cbGenericBreakpoint(BP_TYPE bptype, void* ExceptionAddress = nullptr
     if(*bp.commandText && commandCondition)  //command
     {
         //TODO: commands like run/step etc will fuck up your shit
-        DbgCmdExec(bp.commandText);
+        varset("$breakpointcondition", breakCondition ? 1 : 0, false);
+        varset("$breakpointlogcondition", logCondition, false);
+        _dbg_dbgcmddirectexec(bp.commandText);

```

Comment:
Won't this deadlock if your command is waiting for the user to resume debugging?